scheduler-server/src/main/java/gov/nasa/jpl/aerie/scheduler/server/services/SynchronousSchedulerAgent.java

```diff
@@ -177,14 +179,26 @@ public void schedule(final ScheduleRequest request, final ResultsProtocol.Writer
             .type("SCHEDULING_GOALS_FAILED")
             .message("Scheduling goal%s failed".formatted(failedGoals.size() > 1 ? "s" : ""))
             .data(ResponseSerializers.serializeFailedGoals(failedGoals)));
+        return;
       }
       for (final var compiledGoal : compiledGoals) {
-        final var goal = GoalBuilder
-            .goalOfGoalSpecifier(
-                compiledGoal.getValue(),
-                specification.horizonStartTimestamp(),
-                specification.horizonEndTimestamp(),
-                problem::getActivityType);
+        final Goal goal;
+        try {
+          goal = GoalBuilder
+              .goalOfGoalSpecifier(
+                  compiledGoal.getValue(),
+                  specification.horizonStartTimestamp(),
+                  specification.horizonEndTimestamp(),
+                  problem::getActivityType);
+        } catch (GoalBuilder.GoalBuilderFailure e) {
+          writer.failWith(b -> b
+              .type("SCHEDULING_GOAL_BUILDER_FAILED")
+              .message(e.getMessage())
+              .data(Json.createObjectBuilder()
+                        .add("goal_id", compiledGoal.getKey().id())
+                        .build()));
```

Comment:
You can add a `.trace(e)` at the tail-end of this builder chain to include the stack trace in the failure `reason`.

Not a big deal but eventually we can always revisit this builder pattern and maybe make things a bit more explicit to reduce the chance of an exception's trace being omitted. For example, we could separate this method into `failWithException()` and `failWith()` where `failWithException` must take an exception as an argument.